tensorflow/c/experimental/gradients/math_grad.cc

```diff
@@ -289,6 +293,118 @@ class MulGradientFunction : public GradientFunction {
   vector<AbstractTensorHandle*> forward_inputs;
 };
 
+class Log1pGradientFunction : public GradientFunction {
+ public:
+  explicit Log1pGradientFunction(vector<AbstractTensorHandle*> f_inputs)
+      : forward_inputs(f_inputs) {}
+
+  Status Compute(Context* ctx, const IncomingGradients& grad_inputs,
+                 vector<AbstractTensorHandle*>* grad_outputs) override {
+    // TODO(vnvo2409): Add control dependency
+    /* Given upstream grad U and a Log1p op: Y = log(1 + X), the gradients are:
+     *
+     *    dX = U / (1 + X)
+     *
+     */
+
+    AbstractTensorHandle* upstream_grad = grad_inputs[0];
+    AbstractTensorHandle* X = forward_inputs[0];
+
+    grad_outputs->resize(1);
+    vector<AbstractTensorHandle*> temp_outputs(1);
+
+    // Calculate conjugate of X
+    std::string name = "Conj_Log1p_Grad_X";
+    TF_RETURN_IF_ERROR(
+        Conj(ctx->ctx, {X}, absl::MakeSpan(temp_outputs), name.c_str()));
```

Comment:
But I use `X` here to calculate the conjugate. Am I missing something ?

Reply:
My comment was for `DivNoNanGradientFunction`. There it is unused I think.

Reply:
Sorry. I found it